File: app/agent/local-agent.js
#!/usr/bin/env node
// Read-only example. Use the maintained CLI runner to claim and complete real tasks.
const base = process.env.MENGLABS_API || process.env.API_BASE || 'http://localhost:5173';
const token = process.env.MENGLABS_KEY || process.env.AGENT_API_KEY;
const plan =
	process.env.MENGLABS_PERENCANAAN ||
	process.argv.find((arg) => arg.startsWith('--perencanaan='))?.split('=')[1];
if (!token)
	throw new Error('Set MENGLABS_KEY to a project-scoped token from the Implementasi page.');
const query = new URLSearchParams({ status: 'todo', limit: '5' });
if (plan) query.set('perencanaanId', plan);
const response = await fetch(`${base}/api/agent/tasks?${query}`, {
	headers: { Authorization: `Bearer ${token}` }
});
if (!response.ok)
	throw new Error(`API ${response.status}: ${(await response.text()).slice(0, 300)}`);
console.log(JSON.stringify(await response.json(), null, 2));
console.log(
	'Read-only example: no task was claimed or completed. Use `npx menglabs-ngoding run ...` for execution.'
);

File: app/agent/local-agent.py
#!/usr/bin/env python3
"""Read-only API example. It never claims or completes a task."""
import json
import os
import urllib.parse
import urllib.request

base = os.getenv("MENGLABS_API", os.getenv("API_BASE", "http://localhost:5173"))
token = os.getenv("MENGLABS_KEY", os.getenv("AGENT_API_KEY"))
plan = os.getenv("MENGLABS_PERENCANAAN")
if not token:
    raise SystemExit("Set MENGLABS_KEY to a project-scoped token from the Implementasi page.")
params = {"status": "todo", "limit": "5"}
if plan:
    params["perencanaanId"] = plan
request = urllib.request.Request(f"{base}/api/agent/tasks?{urllib.parse.urlencode(params)}", headers={"Authorization": f"Bearer {token}"})
with urllib.request.urlopen(request, timeout=30) as response:
    print(json.dumps(json.load(response), indent=2))
print("Read-only example: no task was claimed or completed. Use the maintained CLI runner for execution.")
